### ISSUE TITLE
NVDA ARIA alerts omit messages containing HTML

### ISSUE THREAD

**Issue by nvaccessAuto** [nvaccess/nvda#1337]
**Reported by wdmartin on 2011-01-25 23:23**
In Firefox 3.6.13 and NVDA 2010.2, NVDA will only read alerts aloud if they consist of a string that does not contain HTML.  I have built a minimal test case:

http://www.libsys.und.edu/dev/ill/ldap/nvda-test.html

I will also upload a copy of the test case.

The are four ways to update the contents of the alert:

Test 1) Using innerHTML with a plain text string

Test 2) Using innerHTML with a string containing HTML

Test 3) Using the W3C DOM methods to manually create/insert HTML

Test 4) Using the W3C DOM methods to manually insert a plain string

Test 1 is read as "This is test 1 alert This is test 1"

Test 2 is read as "alert"

Test 3 is read as "alert"

Test 4 is read as "This is test 4 alert This is test 4"

This works fine if all you want is an audible message.  But if you want a message which is both audible and visible, it makes things more complicated.

**Comment by leonardder** [comment]
@JCSTeh: Could you restore the attachment to see whether this still occurs

**Comment by jcsteh** [comment]
[nvda-test.html](https://github.com/nvaccess/nvda/files/1157770/nvda-test.html.txt)

**Comment by Adriani90** [comment]
This is working as expected with NVDA 2019.1.1 and Firefox 66.0.3 as well as Chrome 74. All 4 tests are reported by NVDA correctly. In IE this does not work at all. But I guess a fix in IE will highly unlikely.

**Comment by Adriani90** [comment]
Closing as works for me.
